### PR TITLE
tint2: add module

### PR DESCRIPTION
**Pull request by CarlosLoboxyz**
### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

**Review by berbiche — CHANGES_REQUESTED** [review]
Thanks for the PR.

Tint2's settings format seems to be a plain ini file.
Could you add a `settings` option and rename the `config` option to `extraConfig`?

Ideally, the tint2 module should look like the Mercurial one: https://github.com/nix-community/home-manager/blob/4fa1ba72a39da509399cce43995e4c1e1960ce9c/modules/programs/mercurial.nix
That is, it should have a `package`, `settings` and `extraConfig` option.

The module should also declare a systemd service (maybe with an `enable` option?) that will automatically start tint2.

**Comment by CarlosLoboxyz** [comment]
Not sure how test the systemd unit since I'm running xorg using xinit, which means i've to star my user user units manually

File: modules/programs/tint2.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+let
+
+  cfg = config.programs.tint2;
+
+  buildConfig = ''${cfg.settings}'';
+
+in
+
+{
+  meta.maintainers = with maintainers; [ CarlosLoboxyz ];
```
Comment (berbiche) [inline modules/programs/tint2.nix:14]:
```suggestion
  meta.maintainers = [ hm.maintainers.CarlosLoboxyz ];
```

**Review by berbiche — COMMENTED** [review]
You don't need to test the tint2 systemd service if you cannot do it on your system.

**Comment by rycee** [comment]
I made a few minor cleanups and merged to master. I agree that this module really should be a service one and not a program one, hopefully somebody comes along in the future and does the conversion. Can do a `mkRenamedOptionModule` when that happens.

**Pull request merged**